ucfcs/GrowthPlus
# Julio/implement points

This PR implements the scoring system for roadmap one, including flashcards and quizzes, game not implemented yet. 

To test: 
- Wipe out the data and reinstall the app. 
- Create a new parent and then child. 
- Navigate to roadmap one and notice the child starts at tile 1, no others lessons or quizzes should be clickable (game tile still is for now). 
- Start at tile 1 and progress through the roadmap, complete each lesson fully. 
- If lesson is completed and pass successfully, the next tile lesson should unlock, at all times the avatar is at the current lesson(except when all lessons are completed). 
-  As you progress through the lesson flashcards, make sure that scoring in the top bar updates if answer is correct, otherwise it should not update. Flashcard : 2 pt. Quiz question: 1pt.
- Try to fail a flashcard(get less than 3 right), inspect that the next lesson is not unluck, child still gets points for answering a correct flashcard.  
- Go back to previous lesson (already completed), play it as normal, the child score should update but no other events should occur such as the avatar moving or updates to future lessons, avatar only moves by completing current lesson. 
- Get to the quiz tile, play it as normal, fail it(get less than 7 right), the avatar should not move and the next lesson should not unlock.
- Pass the quiz (get more than 6 right), the avatar should move to next lesson and the state updates, continue through the lessons. 
- Try to go back a completed quiz, play it, the only thing that updates is the child score, no other events should occur. 
- Continue with lessons until last lesson.
- Last lesson is all flashcards, fail it (get less than 7 right), make sure avatar is still there. 
- Pass last lesson (get more than 6 right), avatar should not be there and it is now a completed lesson.
